executor/join.go

```diff
@@ -369,6 +376,8 @@ func (e *HashJoinExec) preAlloc4InnerParts() (err error) {
 				return err
 			}
 			if hasNull {
+				// We set the two indexes to MaxUint32 to specify a row with Nil-JoinKey.
+				partPtrs[rowIdx].partitionIdx, partPtrs[rowIdx].rowIdx = math.MaxUint32, math.MaxUint32
```

Comment:
It's better to assign a name to the constant partition address. For example:
```go
const partPtr4NullKey partRowPtr = {math.MaxUint32, math.MaxUint32}
```

and inside this function:
``go
partPtrs[rowIdx] = partPtr4NullKey
```

and in function `doInnerPartition()`:
```go
if partPtr == partPtr4NullKey {
	continue;
}
```